azure-quantum/azure/quantum/optimization/solvers.py

```diff
@@ -95,29 +96,8 @@ def submit(
         # Create a container URL:
         job_id = Job.create_job_id()
         logger.info(f"Submitting job with id: {job_id}")
-
         container_name = f"job-{job_id}"
-
-        if not self.workspace.storage:
-            # No storage account is passed, in this
-            # case, get linked account from the service
-            container_uri = self.workspace._get_linked_storage_sas_uri(
-                container_name
-            )
-            container_client = ContainerClient.from_container_url(
-                container_uri
-            )
-            create_container_using_client(container_client)
-            container_uri = azure.quantum.storage.remove_sas_token(
-                container_uri
-            )
-        else:
-            # Storage account is passed, use it to generate a container_uri
-            container_uri = azure.quantum.storage.get_container_uri(
-                self.workspace.storage, container_name
-            )
-
-        logger.debug(f"Container URI: {container_uri}")
+        container_uri = Job.create_container(self.workspace, container_name)
 
         if isinstance(problem, str):
```

Comment:
Instead of calling `problem.upload`, one thing you could do is call `Job.from_input_data` if the problem is `Problem` and `Job.from_uri`, if using a storage URL, that would remove all the logic of creating the container and job id from this method and pus it to `Job`